### PR TITLE
Fix types generate

### PR DESCRIPTION
**Pull request by huozhi**
### Fix

types generation in subpath module dist files.

in `immutable/dist/immutable/index.d.ts`. but source files won't be included in the published npm package

before
```ts
import { Middleware, SWRHook } from '../src/types';
```

after, import from published module
```ts
import { Middleware, SWRHook } from 'swr';
```

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -5,11 +5,12 @@ module.exports = {
   modulePathIgnorePatterns: ['<rootDir>/examples/'],
   setupFilesAfterEnv: ['<rootDir>/jest-setup.ts'],
   moduleNameMapper: {
-    '^swr$': '<rootDir>/src'
+    '^swr$': '<rootDir>/src',
+    '^@swr-internal/(.*)': '<rootDir>/src/$1'
```
Comment (shuding) [inline jest.config.js:9]: Is `swr-internal` used somewhere?

File: infinite/index.ts [hunk 2]
```diff
@@ -20,7 +18,7 @@ import {
   SWRInfiniteResponse,
   MutatorCallback,
   Middleware
-} from '../src/types'
+} from 'swr'
```
Comment (shuding) [inline infinite/index.ts:21]: Even if we fixed those types, the dependency `import defaultConfig from '../src/utils/config'` will still import types directly from `'../src/types'`. So I feel this is a tricky problem.

Comment (huozhi) [reply infinite/index.ts:21]: Good catch. Seems the types of whole swr core are emitted into package. Need to find a way to get rid of it.

Comment (huozhi) [reply infinite/index.ts:20]: Only include the source files in subpath modules can avoid emitting swr types. But still fews left like `with-middlewares`. We can improve it later by later.

```
infinite/dist/
├── index.esm.js
├── index.js
├── infinite
│   └── index.d.ts
└── src
    └── utils
        ├── env.d.ts
        ├── hash.d.ts
        ├── helper.d.ts
        ├── normalize-args.d.ts
        ├── serialize.d.ts
        └── with-middleware.d.ts

3 directories, 9 files
```

File: infinite/index.ts [hunk 3]
```diff
@@ -20,12 +17,12 @@ import {
...
 
 export const infinite = ((<Data, Error>(useSWRNext: SWRHook) => (
   getKey: KeyLoader<Data>,
   fn: Fetcher<Data> | null,
-  config: typeof defaultConfig & SWRInfiniteConfiguration<Data, Error>
+  config: typeof SWRConfig.default & SWRInfiniteConfiguration<Data, Error>
```
Comment (shuding) [inline infinite/index.ts:25]: Nice one!

File: test/tsconfig.json [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
+  "extends": "../tsconfig.json",
   "compilerOptions": {
-    "rootDir": ".",
+    "rootDir": "..",
     "strict": false
   },
   "include": [
+    "..",
     "./test", 
     "./jest-setup.ts"
```
Comment (shuding) [inline test/tsconfig.json:10]: Maybe:

```suggestion
    "../test", 
    "../jest-setup.ts"
```

File: src/utils/serialize.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import hash from './hash'
-import { Key } from '../types'
+import { Key } from 'swr'
```
Comment (shuding) [inline src/utils/serialize.ts:2]: Feels like a hack, it seems to be working fine though.

File: immutable/index.ts [hunk 6]
```diff
@@ -1,8 +1,5 @@
-// @ts-ignore
-import useSWR from 'swr'
-
+import useSWR, { Middleware, SWRHook } from 'swr'
```
Comment (huozhi) [inline immutable/index.ts:1]: only keep type imports in `immutable` and `infinite`, for reset of the generated types in `subpath/dist/src/xxx/x.d.ts` let's revisit them later

**Review by shuding — APPROVED** [review]

**Pull request merged**